rest/model/conversions.go

```diff
@@ -46,49 +46,73 @@ func conversionFn(opts convertFnOpts, generatedConversions map[typeInfo]string)
 		return simplePtrs(info, generatedConversions)
-	}
-	if intype, inIsPtr := opts.in.(*types.Pointer); inIsPtr {
-		value, isPrimitive := intype.Elem().(*types.Basic)
+	case *types.Pointer:
+		value, isPrimitive := in.Elem().(*types.Basic)
 		if isPrimitive {
 			info.InType = value.String()
 			info.InIsPtr = true
 			return simplePtrs(info, generatedConversions)
 		}
-		underlyingStruct, isStruct := intype.Elem().(*types.Named)
+		underlyingStruct, isStruct := in.Elem().(*types.Named)
 		if isStruct {
 			return objectTypeFnName(underlyingStruct, opts.outIsPtr)
 		}
 		return convertFuncs{}, errors.New("complex pointers not implemented yet")
-	}
-	if intype, inIsNamedType := opts.in.(*types.Named); inIsNamedType {
-		switch intype.String() {
+	case *types.Named:
+		switch in.String() {
 		case "time.Time":
 			return simplePtrs(info, generatedConversions)
 		default:
-			return objectTypeFnName(intype, opts.outIsPtr)
+			return objectTypeFnName(in, opts.outIsPtr)
 		}
-	}
-	if intype, inIsMap := opts.in.(*types.Map); inIsMap {
-		switch intype.String() {
+	case *types.Map:
+		switch in.String() {
 		case "map[string]interface{}":
 			return simplePtrs(info, generatedConversions)
 		default:
 			return convertFuncs{}, errors.New("complex maps not implemented yet")
 		}
-	}
-	if intype, inIsInterface := opts.in.(*types.Interface); inIsInterface {
-		switch intype.String() {
+	case *types.Interface:
+		switch in.String() {
 		case "interface{}":
 			return simplePtrs(info, generatedConversions)
 		default:
 			return convertFuncs{}, errors.New("non-empty interfaces not implemented yet")
 		}
+	case *types.Slice:
+		inName := cleanName(shortenPackage(opts.in.String()), containsPtr(opts.in.String()))
+		outName := cleanName(shortenPackage(opts.outType), containsPtr(opts.outType))
+		return convertFuncs{
+			converter: fmt.Sprintf("%s%s", inName, outName),
+			inverter:  fmt.Sprintf("%s%s", outName, inName),
+		}, nil
+	default:
+		return convertFuncs{}, errors.Errorf("converting type %s is not supported", opts.in.String())
 	}
+}
 
-	return convertFuncs{}, errors.Errorf("converting type %s is not supported", opts.in.String())
+// arrayConversionFn is similar to conversionFn above, but specifically takes an array
+// type and returns the correct function to convert each element of the array, using the
+// naming conventions defined in this file
+func arrayConversionFn(opts convertFnOpts, generatedConversions map[typeInfo]string) (convertFuncs, error) {
+	switch in := opts.in.(type) {
+	case *types.Slice:
+		newOpts := opts
+		newOpts.in = in.Elem()
+		_, newOpts.outType = stripArray(opts.outType)
+		return conversionFn(newOpts, generatedConversions)
+	case *types.Array:
+		newOpts := opts
+		newOpts.in = in.Elem()
+		_, newOpts.outType = stripArray(opts.outType)
+		return conversionFn(newOpts, generatedConversions)
+	default:
+		return convertFuncs{}, errors.New("programmer error: attempted to convert a non-array function")
```

Comment:
Is that the only case left?

Reply:
Should potentially include the type still

Reply:
no, this is a default case that shouldn't ever get hit but has to exist in order to compile